actix/src/sec-5-arbiter.md

```diff
@@ -15,89 +15,148 @@ for your actors to run inside. When you call `start()` on your actor it is then
+> which Node runs) or tokio (upon which Actix is implemented), each Arbiter has
+> a similar processing paradigm.
+
+> Specifically, if you're familiar with how JavaScript handles asynchronicity
+> and how the single-threaded model makes it so JS mostly doesn't need
+> synchronization primitives, this should feel very familiar to you.
+
+One `Arbiter` is in control of one thread with one task queue, and when tasks
+get spawned in an Actor's context or using `Arbiter::spawn`, the task gets
+queued up for execution in that thread.
+
+If you pay close attention to way both `ActorFuture` and the methods in the
+`Actor` trait specify their parameters, actor behavior always references both
+the actor and the context as mutable: `&mut self` (or `&mut A` in `ActorFuture`)
+and `&mut Self::Context`. This is because all of these tasks have **exclusive**
+rights to the actor and its context. This is achieved by putting all actor logic
+in the same thread, so we know no two pieces of logic involving the actor are
+running concurrently without having to use synchronization primitives.
+
+While the library is written with an asynchronous, event-driven flow, and while
+the library otherwise enables concurrency, each Actor specifically runs on one
+thread. Each task gets coordinated by the Arbiter to run eventually, but always
+on the same thread. So when a Future is spawned on the Arbiter (using
+`Arbiter::spawn` or the more common `Context::spawn`), it will execute within
+the same execution context (thread), allowing the Future full access to the
+state while avoiding synchronization (such as `Mutex`es).
+
+When you create a new Arbiter, this creates a new execution context for Actors.
+The new thread is available to add new Actors to it, but Actors cannot freely
+move between Arbiters: they are tied to the Arbiter they were spawned in.
+However, Actors on different Arbiters can still communicate with each other
+using the normal `Addr`/`Recipient` methods. The method of passing messages is
+agnostic to whether the Actors are running on the same or different Arbiters.
+
 ## Using Arbiter for resolving async events
 
-If you aren't an expert in rust futures, Arbiter can be a helpful
-and simple wrapper to resolving async events in order. Consider
-we have two actors, A and B, and we want to run an event on B
-only once a result from A is completed. We can use Arbiter::spawn
-to assist with this task.
+If you aren't an expert in Rust Futures, Arbiter can be a helpful and simple
+wrapper to resolving async events in order. Consider we have two actors, A and
+B, and we want to run an event on B only once a result from A is completed. We
+can use `Arbiter::spawn` to assist with this task.
 
 ```rust
 # extern crate actix;
 # extern crate futures;
-# use futures::Future;
 # use actix::prelude::*;
-# 
-# struct SumActor {}
-# 
-# impl Actor for SumActor {
-#     type Context = Context<Self>;
-# }
+# use futures::Future;
+#
+struct SumActor {}
```

Comment:
I didn't think this part of the example was needed to be displayed given the number of actor examples already?

Reply:
I think the context is just easier to understand when it's visible. I don't feel strongly in this case.

Reply:
@DoumanAsh tends to have strong feelings on this, so I'll ask them to decide :)

Reply:
It is fine

Reply:
Then it's okay by me.